lib/Common/Memory/Recycler.h

```diff
@@ -979,6 +978,10 @@ class Recycler
     PageAllocator backgroundProfilerPageAllocator;
     DListBase<ArenaAllocator> backgroundProfilerArena;
 #endif
+
+    // destruct autoHeap after backgroundProfilerPageAllocator;
+    HeapInfo autoHeap;
```

Comment:
Can you explain why this needs to be after backgroundProfilerPageAllocator.  Those seems to me to be unrelated?